web/source/osk/mouseEventEngine.ts

```diff
@@ -9,27 +9,56 @@ namespace com.keyman.osk {
     private hasActiveClick: boolean = false;
     private ignoreSequence: boolean = false;
 
-    public constructor(vkbd: VisualKeyboard) {
-      // document.body is the event root b/c we need to track the mouse if it leaves
-      // the VisualKeyboard's hierarchy.
-      super(vkbd, document.body);
+    public constructor(config: InputEventEngineConfig) {
+      super(config);
 
       this._mouseStart = this.onMouseStart.bind(this);
       this._mouseMove  = this.onMouseMove.bind(this);
       this._mouseEnd   = this.onMouseEnd.bind(this);
```

Comment:
I take it mouseEventEngine.ts currently only handles left mouse clicks. Could a future feature allow right-click (on a suggestion) for a "copy" option? Just wondering if later on we would add right and middle clicks...

Reply:
Maybe?  It'd definitely be a "future feature", though.

We'd probably want to include something similar for touch - I know that simultaneous double-touch and triple-touch act as analogies for the other mouse clicks on some devices, so it wouldn't be unreasonable to extend the abstraction to incorporate that if desired.